javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/IndyTypeTransformerImpl.java

```diff
@@ -30,7 +30,7 @@ public IndyTypeTransformerImpl(
     this.instrumentationModule = module;
     this.adviceMapping =
         Advice.withCustomMapping()
-            .with(new Advice.AssignReturned.Factory().withSuppressed(Throwable.class))
+            .with(new ForceDynamicallyTypedAssignReturnedFactory())
```

Comment:
wouldn't it work if you add both of these instead of making `ForceDynamicallyTypedAssignReturnedFactory` delegate to the other one?

Reply:
You mean adding both the `Advice.AssignReturned.Factory().withSuppressed(Throwable.class)` and `ForceDynamicallyTypedAssignReturnedFactory()`?

I don't think that would work, because then byteBuddy would try to process each annotation twice I think.

What I could do is add a `withSuppressed` method to `ForceDynamicallyTypedAssignReturnedFactory`, which delegates the call to the wrapped `Advice.AssignReturned.Factory()` underneath.

Alternatively the `ForceDynamicallyTypedAssignReturnedFactory` constructor could accept the instance of the `Advice.AssignReturned.Factory()` which it should wrap around.

We need to perform wrapping because `ForceDynamicallyTypedAssignReturnedFactory` changes the `MethodDescription` by modifying the annotations and that changed method description needs to be passed to the factory performing the actual bytecode generation.

Reply:
> Alternatively the ForceDynamicallyTypedAssignReturnedFactory constructor could accept the instance of the Advice.AssignReturned.Factory() which it should wrap around.

I think this is the best option.